pkg/mcs/scheduling/server/server.go

```diff
@@ -479,7 +479,7 @@ func (s *Server) startWatcher() (err error) {
 	if err != nil {
 		return err
 	}
-	s.ruleWatcher, err = rule.NewWatcher(s.Context(), s.GetClient(), s.clusterID)
+	s.ruleWatcher, err = rule.NewWatcher(s.Context(), s.GetClient(), s.clusterID, s.storage)
```

Comment:
BTW, after the change of placement rules, we also need to call `AddSuspectKeyRange`, which is similar to `/config/rule [post]`, `/config/rule/{group}/{id} [delete]`, `/config/rule_group [post]`, `/config/rule_group/{id} [delete]`

Reply:
I will implement this in a separate PR.

Reply:
Implemented in https://github.com/tikv/pd/pull/7213.